app/assets/javascripts/components/course_creator/course_creator.jsx

```diff
@@ -67,11 +67,9 @@ const CourseCreator = createReactClass({
   },
 
   UNSAFE_componentWillReceiveProps(nextProps) {
-    if (nextProps.course.school !== '' && nextProps.course.title !== '') {
-      this.state.tempCourseId = CourseUtils.generateTempId(nextProps.course);
-    } else {
-      this.state.tempCourseId = '';
-    }
+    this.setState({
```

Comment:
I think this may result in behavior change when creating a new course.

The idea is that we don't want the current version of the course slug (ie, the tempCourseId) to show up in the CourseCreator modal at the bottom until after the user has added something for both school and title.

Reply:
Nevermind! I see that you moved that behavior into CourseUtils. Good!